streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -223,7 +223,11 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
     // Signal the receivers to delete old block data
     if (WriteAheadLogUtils.enableReceiverLog(ssc.conf)) {
       logInfo(s"Cleanup old received batch data: $cleanupThreshTime")
-      endpoint.send(CleanupOldBlocks(cleanupThreshTime))
```

Comment:
I saw NullPointerException once when I ran the tests a lot of times. The `JobScheduler.eventLoop` is stopped after `ReceiverTracker.stop` and `ReceiverTracker.stop` sets `endpoint` to null. Therefore it's possible that this method is called between `ReceiverTracker.stop` and `JobScheduler.eventLoop.stop` and will throw NullPointerException.